### DISCUSSION TITLE
Run Gauge with Poetry

### DISCUSSION THREAD

**Discussion by nagamanickamm** [getgauge/gauge#2263] [General]
Hello!

I have configured Poetry for my Gauge project, switched the Python Interrupter to Poetry environment.
I see no interpreter error however if I run from Test Runner or from Command I see import errors.
Even tried to manually activate to my virtual environment (Poetry) but still facing issue.

Can someone advise how to do this?

8-07-2022 15:23:42.268 [Gauge] [ERROR] Error occurred while waiting for runner process to finish.
Error : exit status 1
18-07-2022 15:23:42.268 [Gauge] [CRITICAL] Failed to start gauge API: Error occurred while waiting for runner process to finish.
Error : exit status 1
18-07-2022 15:26:24.297 [Gauge] [DEBUG] Created gauge_screenshots_dir at C:\IGS\GitHub\NeonTestAutomation\demo\.gauge\screenshots
18-07-2022 15:26:24.298 [Gauge] [DEBUG] Plugin python is already installed.
18-07-2022 15:26:24.299 [Gauge] [DEBUG] Plugin html-report is already installed.
18-07-2022 15:26:24.299 [Gauge] [DEBUG] Plugin json-report is already installed.
18-07-2022 15:26:24.300 [Gauge] [DEBUG] Plugin xml-report is already installed.
18-07-2022 15:26:24.300 [Gauge] [DEBUG] Plugin spectacle is already installed.
18-07-2022 15:26:24.301 [Gauge] [DEBUG] Plugin flash is already installed.
18-07-2022 15:26:24.301 [Gauge] [DEBUG] Parsing started.
18-07-2022 15:26:24.301 [Gauge] [DEBUG] Started concepts parsing.
18-07-2022 15:26:24.304 [Gauge] [DEBUG] 0 concepts parsing completed.
18-07-2022 15:26:24.304 [Gauge] [DEBUG] Started specifications parsing.
18-07-2022 15:26:24.304 [Gauge] [DEBUG] Reading data file: C:\IGS\GitHub\NeonTestAutomation\demo\testLibrary\resources\login_testdata.csv
18-07-2022 15:26:24.305 [Gauge] [DEBUG] 1 specifications parsing completed.
18-07-2022 15:26:24.305 [Gauge] [DEBUG] Parsing completed.
18-07-2022 15:26:24.866 [Gauge] [DEBUG] Checking updates...
18-07-2022 15:26:24.869 [Gauge] [DEBUG] Downloading https://downloads.gauge.org/plugin/flash?l=python&p=flash,html-report,json-report,python,screenshot,spectacle,xml-report&o=windows&a=amd64
18-07-2022 15:26:24.999 [Gauge] [DEBUG] Downloading https://downloads.gauge.org/plugin/html-report?l=python&p=flash,html-report,json-report,python,screenshot,spectacle,xml-report&o=windows&a=amd64
18-07-2022 15:26:25.055 [Gauge] [DEBUG] Downloading https://downloads.gauge.org/plugin/json-report?l=python&p=flash,html-report,json-report,python,screenshot,spectacle,xml-report&o=windows&a=amd64
18-07-2022 15:26:25.064 [python] [ERROR] Traceback (most recent call last):
18-07-2022 15:26:25.064 [python] [ERROR] File "C:\Users\naga.mani\AppData\Roaming\gauge\plugins\python\0.3.17\start.py", line 11, in <module>
18-07-2022 15:26:25.064 [python] [ERROR] from getgauge import handlers, logger, processor
18-07-2022 15:26:25.064 [python] [ERROR] File "C:\Users\naga.mani\AppData\Local\pypoetry\Cache\virtualenvs\neontestautomation-bXgG8QID-py3.10\lib\site-packages\getgauge\handlers.py", line 4, in <module>
18-07-2022 15:26:25.064 [python] [ERROR] from getgauge import logger, processor
18-07-2022 15:26:25.064 [python] [ERROR] File "C:\Users\naga.mani\AppData\Local\pypoetry\Cache\virtualenvs\neontestautomation-bXgG8QID-py3.10\lib\site-packages\getgauge\processor.py", line 7, in <module>
18-07-2022 15:26:25.064 [python] [ERROR] from getgauge.executor import (create_execution_status_response,
18-07-2022 15:26:25.064 [python] [ERROR] File "C:\Users\naga.mani\AppData\Local\pypoetry\Cache\virtualenvs\neontestautomation-bXgG8QID-py3.10\lib\site-packages\getgauge\executor.py", line 7, in <module>
18-07-2022 15:26:25.065 [python] [ERROR] from getgauge.messages.messages_pb2 import ExecutionStatusResponse, Message
18-07-2022 15:26:25.065 [python] [ERROR] File "C:\Users\naga.mani\AppData\Local\pypoetry\Cache\virtualenvs\neontestautomation-bXgG8QID-py3.10\lib\site-packages\getgauge\messages\messages_pb2.py", line 6, in <module>
18-07-2022 15:26:25.065 [python] [ERROR] import getgauge.messages.spec_pb2 as spec__pb2
18-07-2022 15:26:25.065 [python] [ERROR] File "C:\Users\naga.mani\AppData\Local\pypoetry\Cache\virtualenvs\neontestautomation-bXgG8QID-py3.10\lib\site-packages\getgauge\messages\spec_pb2.py", line 34, in <module>
18-07-2022 15:26:25.065 [python] [ERROR] _descriptor.EnumValueDescriptor(
18-07-2022 15:26:25.065 [python] [ERROR] File "C:\Users\naga.mani\AppData\Local\pypoetry\Cache\virtualenvs\neontestautomation-bXgG8QID-py3.10\lib\site-packages\google\protobuf\descriptor.py", line 755, in __new__
18-07-2022 15:26:25.065 [python] [ERROR] _message.Message._CheckCalledFromGeneratedFile()
18-07-2022 15:26:25.066 [python] [ERROR] TypeError: Descriptors cannot not be created directly.
18-07-2022 15:26:25.066 [python] [ERROR] If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
18-07-2022 15:26:25.066 [python] [ERROR] If you cannot immediately regenerate your protos, some other possible workarounds are:
18-07-2022 15:26:25.067 [python] [ERROR] 1. Downgrade the protobuf package to 3.20.x or lower.
18-07-2022 15:26:25.067 [python] [ERROR] 2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
18-07-2022 15:26:25.067 [python] [ERROR] More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
18-07-2022 15:26:25.114 [Gauge] [ERROR] Error occurred while waiting for runner process to finish.
Error : exit status 1
18-07-2022 15:26:25.114 [Gauge] [CRITICAL] Failed to start gauge API: Error occurred while waiting for runner process to finish.
Error : exit status 1

**Comment by zabil** [comment]
If you are using poetry try running gauge with command

```
poetry run gauge run
```
